tests/test_netcdf_reader.py

```diff
@@ -0,0 +1,50 @@
+from unittest.mock import patch
+
+import pytest
+import xarray as xr
+
+from nsidc.metgen import netcdf_reader
+
+# Unit tests for the 'netcdf_reader' module functions.
+#
+# The test boundary is the netcdf_reader module's interface with the filesystem
+# so in addition to testing the netcdf_reader module's behavior, the tests
+# should mock those module's functions and assert that netcdf_reader functions
+# call them with the correct parameters, correctly handle their return values,
+# and handle any exceptions they may throw.
+
+@pytest.fixture
+def netcdf_file():
+    x = [100, 200, 300]
+    y = [100, 200, 300]
+    return xr.Dataset(coords={"x": (["loc"], x), "y": (["loc"], y)})
+
+@patch('nsidc.metgen.netcdf_reader.xr.open_dataset', return_value = netcdf_file)
```

Comment:
Fails! (As does `netcdf_file()` because fixtures can't be called directly). Is there a way to mock the xarray object? This may be a clue that the code needs some refactoring.

Reply:
What I'll likely do is not worry about a direct test of `extract_metadata`, and will just go with tests of the other methods in this file. I'm still interested in the answer to this question, though!

Reply:
I think the only way to use the fixture in the mock is to get rid of the @patch decorator and instead create the patch in code inside the test function. So add the `netcdf_file` fixture to the unit test function signature, then write a line of code inside of it which does something like `mock_dataset = patch(...)`. See: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch